Nukkit tick error spams console

2016-4-8 09:16:44 [INFO] Loading nukkit.yml...
2016-4-8 09:16:44 [INFO] Loading server properties...
2016-4-8 09:16:44 [INFO] Selected English (eng) as the base language
2016-4-8 09:16:44 [INFO] Starting Minecraft: PE server version v0.14.0 alpha
2016-4-8 09:16:44 [INFO] Opening server on 0.0.0.0:19180
2016-4-8 09:16:44 [INFO] This server is running Nukkit version 1.0dev "蘋果(Apple)派(Pie)" (API 1.0.0)
2016-4-8 09:16:44 [INFO] Nukkit is distributed under the LGPL License
2016-4-8 09:16:44 [INFO] Loading AnyVersion v0.0.2
2016-4-8 09:16:44 [INFO] Preparing level "Arogah - by Zwoeard"
2016-4-8 09:16:44 [INFO] Enabling AnyVersion v0.0.2
2016-4-8 09:16:44 [INFO] [AnyVersion] Server protocol version: 45
2016-4-8 09:16:44 [INFO] Starting GS4 status listener
2016-4-8 09:16:44 [INFO] Setting query port to 19180
2016-4-8 09:16:44 [INFO] Query running on 0.0.0.0:19180
2016-4-8 09:16:44 [INFO] Default game type: Survival Mode
2016-4-8 09:16:44 [INFO] Done (0.586s)! For help, type "help" or "?"
2016-4-8 09:16:51 [INFO] [AnyVersion] FuryTacticzHD is using protocol version 46 (server version: 45)
2016-4-8 09:16:51 [INFO] [AnyVersion] Warning! Using outdated/outrunning client could damage your server.
2016-4-8 09:16:51 [INFO] [AnyVersion] Use it on your own risk
2016-4-8 09:16:51 [INFO] FuryTacticzHD[/F OFF THATS MY IP:19132] logged in with entity id 1 at (Arogah - by Zwoeard, 163.0481, 72.0, 145.0194)
2016-4-8 09:16:52 [INFO] FuryTacticzHD joined the game
2016-4-8 09:17:07 [INFO] [FuryTacticzHD: Teleported FuryTacticzHD to 600.0, 128.0, 700.0]
2016-4-8 09:17:37 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:17:37 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:17:37 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:17:37 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:05 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:05 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:13 [INFO] [FuryTacticzHD: Set the world spawn point to (555.00, 85.00, 309.00)]
2016-4-8 09:18:16 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:16 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:20 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:20 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:21 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:21 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:29 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:29 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

2016-4-8 09:18:29 [ERROR] Error while unloading a chunk: java.lang.RuntimeException: java.lang.NullPointerException
2016-4-8 09:18:29 [ALERT] java.lang.RuntimeException: java.lang.NullPointerException
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:276)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2340)
	at cn.nukkit.level.Level.unloadChunk(Level.java:2304)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2584)
	at cn.nukkit.level.Level.unloadChunks(Level.java:2561)
	at cn.nukkit.level.Level.doTick(Level.java:670)
	at cn.nukkit.Server.checkTickUpdates(Server.java:903)
	at cn.nukkit.Server.tick(Server.java:965)
	at cn.nukkit.Server.tickProcessor(Server.java:768)
	at cn.nukkit.Server.start(Server.java:748)
	at cn.nukkit.Server.<init>(Server.java:445)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
Caused by: java.lang.NullPointerException
	at cn.nukkit.entity.item.EntityItem.saveNBT(EntityItem.java:181)
	at cn.nukkit.level.format.anvil.Chunk.toBinary(Chunk.java:325)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:143)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:274)
	... 11 more

--- a/src/main/java/cn/nukkit/entity/item/EntityItem.java
+++ b/src/main/java/cn/nukkit/entity/item/EntityItem.java
@@ -176,6 +176,9 @@ public class EntityItem extends Entity {
 
     @Override
     public void saveNBT() {
+    	if(this.item == null){
+    		return;
+    	}
         super.saveNBT();
         this.namedTag.putCompound("Item", new CompoundTag()
                 .putShort("id", this.item.getId())

--- a/src/main/java/cn/nukkit/event/entity/EntityPortalEnterEvent.java
+++ b/src/main/java/cn/nukkit/event/entity/EntityPortalEnterEvent.java
@@ -1,9 +1,10 @@
 package cn.nukkit.event.entity;
 
 import cn.nukkit.entity.Entity;
+import cn.nukkit.event.Cancellable;
 import cn.nukkit.event.HandlerList;
 
-public class EntityPortalEnterEvent extends EntityEvent {
+public class EntityPortalEnterEvent extends EntityEvent implements Cancellable {
     private static final HandlerList handlers = new HandlerList();
 
     public static final int TYPE_NETHER = 0;

